javascript/javascript2/week1/SpiritNameGen.js

```diff
@@ -0,0 +1,34 @@
+const myButton = document.querySelector("button");
+let inputText;
+const spiritAnimalArray = [
+  "The Icy Bison",
+  "The Swift Armadillo",
+  "The Miniature Buffalo",
+  "The Capable Quail",
+  "The Gleaming Pig",
+  "The Enthusiastic Crane",
+  "The Amusing Penguin",
+  "The Sparkling Reindeer",
+  "The Expensive Caterpillar",
+  "The Intelligent Pony",
+];
+
+myButton.onclick = function (event) {
+  if (inputText.length > 0) {
+    const section = document.querySelector("section");
+    const p = document.createElement("p");
+    const randomNumber = Math.floor(Math.random() * spiritAnimalArray.length);
+    p.innerText = inputText + " - " + spiritAnimalArray[randomNumber];
+    section.appendChild(p);
+  } else {
+    const section = document.querySelector("section");
+    const p = document.createElement("p");
+    p.innerText = "Please input your name to generate your spirit animal.";
+    section.appendChild(p);
+  }
+};
+
+const myInput = document.querySelector("input");
```

Comment:
In here I think you just forgot to add the functionality you have also in the code above. 

oninput  you just assign new value to the inputText but you don't actually use it. 

```